lib/collection/src/common/eta_calculator.rs

```diff
@@ -0,0 +1,71 @@
+use std::time::{Duration, Instant};
+
+/// A progress ETA calculator.
+/// Calculates the ETA roughly based on the last ten seconds of measurements.
+pub struct EtaCalculator {
```

Comment:
I find the ring based design a bit difficult to review TBH

It is fine that you picked this, it is your call :+1: 

In that case I'd love a few unit tests for it because there is a lot of array indexing manipulations going on.

Reply:
Also, would it be useful to utilize any of the [`ringbuffer`](https://docs.rs/ringbuffer/)s so we don't have to manage it ourselves?

Reply:
Added a dependency on `ringbuffer` and some commentary so it should be more clear.
Also, added a test.